opensoundscape/preprocess/preprocessors.py

```diff
@@ -138,23 +140,34 @@ def forward(
             "_start_time": None if clip_times is None else clip_times["start_time"],
             "_sample_duration": self.sample_duration,
             "_preprocessor": self,
+            "_trace": self.pipeline.copy(deep=True) if trace else None,
         }
 
         try:
             # perform each action in the pipeline
             for k, action in self.pipeline.items():
                 if type(action) == break_on_type or k == break_on_key:
+                    if trace:
+                        sample_info["_trace"][
+                            k
+                        ] = f"## Pipeline terminated ## {sample_info['_trace'][k]}"
```

Comment:
so it looks like actions after a terminated pipeline would still contain the Action (copied from the original pipeline in L143. I think a better approach would be to just create a dictionary of the {pipeline_keys: outputs} rather than creating a deepcopy of the pipeline. (Since this sample_info will get passed around, potentially across devices like cpu/gpu, passing a copied object is unnecessarily heavy).